shared/modules/fetch-with-timeout.ts

```diff
@@ -17,12 +30,9 @@ const getFetchWithTimeout = memoize((timeout = SECOND * 30) => {
     const timer = setTimeout(() => abortController.abort(), timeout);
 
     try {
-      const res = await f;
+      return await f;
+    } finally {
       clearTimeout(timer);
-      return res;
-    } catch (e) {
-      clearTimeout(timer);
```

Comment:
I was getting a warning while running the Jest tests for this file that a worker process failed to finish. It seemed to somehow be related to the timer not being totally cleared in this failure case. Since we call `clearTimeout` in both the `try` and `catch` branches, it seemed like we could move it to a `finally`.